### PR TITLE
Add Tanzania holidays

### LINKED ISSUE #1293: Add United Republic of Tanzania holidays
Links:
  - https://en.wikipedia.org/wiki/Public_holidays_in_Tanzania
  - https://www.timeanddate.com/holidays/tanzania/

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

Add Tanzania holidays (`en_US`, **`sw`** localization; `BANK`, **`PUBLIC`** category) from 1994 onwards, with exact Islamic holiday dates from 2013-2023.

Closes #1293.

## Type of change

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/tanzania.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        # Sheikh Abeid Amant Karume.
+        # Status: In-Use.
+        # Commemorates the death of Abeid Amani Karume on Apr 7, 1972.
+        # Assumed to start via Government Notices 79 of 1977.
+
+        self._add_holiday_apr_7(
+            # The Sheikh Abeid Amani Karume Day.
+            tr(
+                "Siku ya kumbukumbu ya Rais wa Kwanza wa Serikali ya Mapinduzi Zanzibar "
+                "Sheikh Abeid Amant Karume"
```
Comment (KJhellico) [inline holidays/countries/tanzania.py:144]:
```suggestion
                "Sheikh Abeid Amani Karume"
```

**Review by KJhellico — COMMENTED** [review]
That's very good! Just one little typo to fix.

File: holidays/countries/tanzania.py (hunk 1 above)
Comment (PPsyrius) [reply holidays/countries/tanzania.py:144]: Oof, missed - this is hopefully fixed now.

File: holidays/countries/tanzania.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+from holidays.groups import ChristianHolidays, InternationalHolidays, IslamicHolidays
+from holidays.holiday_base import HolidayBase
+
+
+class Tanzania(
+    HolidayBase,
+    ChristianHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+):
```
Comment (arkid15r) [inline holidays/countries/tanzania.py:26]:
```suggestion
class Tanzania(HolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
```

File: holidays/countries/tanzania.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    - https://www.bot.go.tz/webdocs/Other/2023%20public%20holidays.pdf  # 2023
+    - https://www.bot.go.tz/webdocs/Other/PUBLIC%20HOLIDAYS%202022.pdf  # 2022
+    - https://www.bot.go.tz/webdocs/Other/PUBLIC%20HOLIDAYS%202021.pdf  # 2021
+    - https://www.bot.go.tz/webdocs/Other/PUBLIC%20HOLIDAYS%202020.pdf  # 2020
+    - https://issamichuzi.blogspot.com/2017/11/sikukuu-za-kitaifa-zenye-mapumziko-kwa.html  # 2018
+    - https://www.timeanddate.com/holidays/tanzania/ (from 2013 onwards)
+
+    Limitations:
+
+    - Only works from 1994 onwards due to the lack of sources for certain legislations, which are:
```
Comment (arkid15r) [inline holidays/countries/tanzania.py:50]:
```suggestion
    - Only works from 1994 onwards due to the lack of sources for certain legislation:
```

File: holidays/countries/tanzania.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+        Government Notices No. 79 of 1977
+        Government Notices No. 300 of 1985
+        Government Notices No. 296 of 1994
+
+    - Exact Islamic holidays date are only available from 2013-2023; the rest being estimates.
```
Comment (arkid15r) [inline holidays/countries/tanzania.py:55]:
```suggestion
    - Exact Islamic holidays dates are only available for 2013-2023; the rest are estimates.
```

File: holidays/countries/tanzania.py [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+        self._add_easter_sunday(tr("Sikukuu ya Pasaka"))
+
+    def _populate_public_holidays(self):
+        # Written Law (Miscellaneous Amendments) (No. 2) Act, 1994.
+        if self._year <= 1993:
+            return None
+
+        # In-lieus ("Badala ya %s") are observed on Monday should it fall on the weekends.
+        # Abrogated in Public Holidays Ordinance  No. 28 of 1966.
+        # Reinstitued in Written Law (Miscellaneous Amendments) (No. 2) Act, 1994.
```
Comment (arkid15r) [inline holidays/countries/tanzania.py:114]:
```suggestion
        # Reinstituted in Written Law (Miscellaneous Amendments) (No. 2) Act, 1994.
```

File: holidays/countries/tanzania.py [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+        # Claimed to be abrogated again on an unspecified date.
+
+        # Fixed Date Holidays.
+
+        # Mwaka Mpya.
+        # Status: In-Use.
+        # Abrogated in Public Holidays Ordinance No. 28 of 1966.
+        # Reinstitued prior to the Written Law (Miscellaneous Amendments) (No. 1) Act, 1993.
```
Comment (arkid15r) [inline holidays/countries/tanzania.py:122]:
```suggestion
        # Reinstituted prior to the Written Law (Miscellaneous Amendments) (No. 1) Act, 1993.
```

File: tests/countries/test_tanzania.py [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+            ("2023-07-07", "Sabasaba"),
+            ("2023-08-08", "Siku ya Wakulima"),
+            ("2023-09-28", "Maulidi"),
+            ("2023-10-14", "Kumbukumbu ya Mwalimu Nyerere"),
+            ("2023-12-09", "Uhuru na Jamhuri"),
+            ("2023-12-25", "Kuzaliwa Kristo"),
+            ("2023-12-26", "Siku ya Kupeana Zawadi"),
+        )
+
+    def test_l10n_en_US(self):
```
Comment (arkid15r) [inline tests/countries/test_tanzania.py:181]:
```suggestion
    def test_l10n_en_us(self):
```

**Review by arkid15r — COMMENTED** [review]
Very nice, another country holidays support! Great work, @PPsyrius!

**Review by arkid15r — APPROVED** [review]
🇹🇿 LGTM

**Pull request merged**